src/Internal/Contract/Monad.purs

```diff
@@ -0,0 +1,400 @@
+  ( class MonadError
+  , class MonadThrow
+  , throwError
+  )
+import Control.Monad.Logger.Class (class MonadLogger)
+import Control.Monad.Reader.Class (class MonadAsk, class MonadReader, ask, asks)
+import Control.Monad.Reader.Trans (ReaderT, runReaderT, withReaderT)
+import Control.Monad.Rec.Class (class MonadRec)
+import Control.Parallel
+  ( class Parallel
+  , parTraverse
+  , parTraverse_
+  , parallel
+  , sequential
+  )
+import Control.Plus (class Plus)
+import Ctl.Internal.Cardano.Types.Transaction (UtxoMap)
+import Ctl.Internal.Contract.Hooks (Hooks)
+import Ctl.Internal.Contract.QueryBackend
+  ( CtlBackend
+  , QueryBackend(BlockfrostBackend, CtlBackend)
+  , QueryBackendLabel(CtlBackendLabel)
+  , QueryBackendParams(BlockfrostBackendParams, CtlBackendParams)
+  , QueryBackends
+  , defaultBackend
+  , lookupBackend
+  )
+import Ctl.Internal.Helpers (filterMapWithKeyM, liftM, logWithLevel)
+import Ctl.Internal.JsWebSocket (_wsClose, _wsFinalize)
+import Ctl.Internal.Logging (Logger, mkLogger, setupLogs)
+import Ctl.Internal.QueryM
+  ( QueryEnv
+  , QueryM
+  , WebSocket
+  , getProtocolParametersAff
+  , getSystemStartAff
+  , mkDatumCacheWebSocketAff
+  , mkOgmiosWebSocketAff
+  , underlyingWebSocket
+  )
+-- TODO Move/translate these types into Cardano
+import Ctl.Internal.QueryM.Ogmios
+  ( ProtocolParameters
+  , SystemStart
+  ) as Ogmios
+import Ctl.Internal.QueryM.ServerConfig (ServerConfig)
+import Ctl.Internal.Serialization.Address (NetworkId)
+import Ctl.Internal.Types.UsedTxOuts (UsedTxOuts, isTxOutRefUsed, newUsedTxOuts)
+import Ctl.Internal.Wallet (Wallet)
+import Ctl.Internal.Wallet (getNetworkId) as Wallet
+import Ctl.Internal.Wallet.Spec (WalletSpec, mkWalletBySpec)
+import Data.Either (Either(Left, Right))
+import Data.Log.Level (LogLevel)
+import Data.Log.Message (Message)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Data.Newtype (class Newtype, unwrap)
+import Data.Traversable (for_, traverse)
+import Effect (Effect)
+import Effect.Aff (Aff, ParAff, attempt, error, finally, supervise)
+import Effect.Aff.Class (liftAff)
+import Effect.Class (class MonadEffect, liftEffect)
+import Effect.Exception (Error, throw, try)
+import Effect.Ref (new) as Ref
+import MedeaPrelude (class MonadAff)
+import Record.Builder (build, merge)
+import Undefined (undefined)
+
+--------------------------------------------------------------------------------
+-- Contract
+--------------------------------------------------------------------------------
+
+-- | The `Contract` monad is a newtype wrapper over `ReaderT` on `ContractEnv`
+-- | over asynchronous effects, `Aff`. Throwing and catching errors can
+-- | therefore be implemented with native JavaScript `Effect.Exception.Error`s
+-- | and `Effect.Class.Console.log` replaces the `Writer` monad. `Aff` enables
+-- | the user to make effectful calls inside this `Contract` monad.
+newtype Contract (a :: Type) = Contract (ReaderT ContractEnv Aff a)
```

Comment:
Contract no longer has `Alt` or `Plus` instances, is this intended? If so we should remember to add it to the changelog

Reply:
Why no `Alt` anymore?

Reply:
Oversight, this has been added back